tests/chainer_tests/functions_tests/math_tests/test_prod.py

```diff
@@ -23,65 +19,47 @@
     'contain_zero': [False],
     'shape': [(3, 1, 0)],
 })))
-class TestProd(unittest.TestCase):
+@testing.fix_random()
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [
+        {},
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'cuda_device': [0, 1],
+    })
+    # ChainerX tests
+    + [
+        {'use_chainerx': True, 'chainerx_device': 'native:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+    ])
+class TestProd(testing.FunctionTestCase):
 
     def setUp(self):
-        self.x = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
-        if self.contain_zero:
-            index = numpy.random.choice(self.x.size)
-            self.x.ravel()[index] = 0
-        g_shape = self.x.prod(axis=self.axis, keepdims=self.keepdims).shape
-        self.gy = numpy.random.uniform(-1, 1, g_shape).astype(self.dtype)
-        self.ggx = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
-
-    def check_forward(self, x_data):
-        x = chainer.Variable(x_data)
-        y = functions.prod(x, axis=self.axis, keepdims=self.keepdims)
-        self.assertEqual(y.data.dtype, self.dtype)
-        y_expect = self.x.prod(axis=self.axis, keepdims=self.keepdims)
-
         if self.dtype == numpy.float16:
-            options = {'atol': 1e-3, 'rtol': 1e-3}
-        else:
-            options = {}
-
-        testing.assert_allclose(y_expect, y.data, **options)
-
-    def test_forward_cpu(self):
-        self.check_forward(self.x)
-
-    @attr.gpu
-    def test_forward_gpu(self):
-        self.check_forward(cuda.to_gpu(self.x))
-
-    def check_backward(self, x_data, y_grad):
-        def f(x):
-            return functions.prod(x, axis=self.axis, keepdims=self.keepdims)
+            self.check_forward_options.update({'atol': 1e-3, 'rtol': 1e-3})
+            self.check_backward_options.update({'atol': 1e-3, 'rtol': 1e-3})
+            self.check_double_backward_options \
+                .update({'atol': 1e-3, 'rtol': 1e-3})
```

Comment:
Could you avoid `\` for line continuation?

https://www.python.org/dev/peps/pep-0008/#maximum-line-length

> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.